src/box/txn.c

```diff
@@ -1280,3 +1322,11 @@ txn_attach(struct txn *txn)
 	trigger_add(&fiber()->on_yield, &txn->fiber_on_yield);
 	trigger_add(&fiber()->on_stop, &txn->fiber_on_stop);
 }
+
+void
+txn_set_timeout(double timeout)
+{
+	struct txn *txn = in_txn();
+	if (txn != NULL)
+		txn->timeout = timeout;
+}
```

Comment:
txn_* functions usually take struct txn as a first argument. Let's make txn_set_timeout take struct txn as well and make it inline, because it's a one-liner.

For Lua and public C API I think we should add box_txn_set_timeout wrapper.

Reply:
Fixed